src/transaction-runner.coffee

```diff
@@ -450,12 +450,16 @@ class TransactionRunner
       hostname: transaction.host
       port: transaction.port
 
-    return {
+    options =
       uri: url.format(urlObject) + transaction.fullPath
       method: transaction.request.method
       headers: transaction.request.headers
       body: transaction.request.body
-    }
+
+    if @configuration.options.proxy
```

Comment:
NIT: you are using `options.proxy = @configuration.options.proxy if @configuration.options.proxy` at https://github.com/apiaryio/dredd/pull/749/commits/02633099e2f070654fcf313335a5347ceec26545#diff-4fe94b7f756f84c61823b246b6738096R156, so you can unify the style, but not big deal.